app/models/notification.rb

```diff
@@ -120,20 +120,20 @@ def send_milestone_notification_without_delay(type:, article_id:)
       Notifications::MilestoneJob.perform_now(type, article_id)
     end
 
-    def remove_all(notifiable_id:, notifiable_type:, action: nil)
-      Notifications::RemoveAllJob.perform_later(notifiable_id, notifiable_type, action)
+    def remove_all_by_action(notifiable:, action: nil)
+      Notifications::RemoveAllByActionJob.perform_later(notifiable, action)
```

Comment:
The job sehould receive a pair of id and type, like in other job classes the job itself should load the object from the DB and do nothing if the object doesn't exist. This avoid deserialization errors and the corner case of the notifiable that has been deleted before this job is called